common/djangoapps/student/views.py

```diff
@@ -703,13 +703,34 @@ def dashboard(request):
     course_entitlement_available_sessions = {}
     for course_entitlement in course_entitlements:
         course_entitlement.update_expired_at()
-        # Filter only the course runs that do not have an enrollment_end date set, or have one set in the future
         course_runs_for_course = get_course_runs_for_course(str(course_entitlement.course_uuid))
         enrollable_course_runs = []
 
+        # Only show published course runs that can still be enrolled and upgraded
+        now = datetime.datetime.now(UTC)
```

Comment:
Can you pull all this into a single method to get the enrollable_course_runs?  It will make it easier to refactor later if we need to.